### ISSUE TITLE
Invalid negative value for <rect> attribute (Initial width value)

### ISSUE THREAD

**Issue by irvinebroque** [luciddg/building-os-charts#5]
If you pass `<TimeseriesChart>` a width of 0, you end up with lots of these errors when working with building-os-charts:

``` javascript
Error: Invalid negative value for <rect> attribute width="-21” 
```

Width = 0 sounds crazy, but with a responsive width container, in React, we end up with code like this, where the width of the container is not known until after the initial render:

``` javascript
module.exports = React.createClass({

  getInitialState() {
    return {
      contentWidth: 0
    };
  },

  componentDidMount() {
    window.addEventListener('resize', this.handleResize);
    var contentNode = React.findDOMNode(this.refs.content);
    this.setState({ contentWidth: contentNode.clientWidth });
  },

  componentWillUnmount() {
    window.removeEventListener('resize', this.handleResize);
  },

  render() {
    <div ref="content">
      <TimeseriesChart
        height={160}
        groups={groups}
        margins={21}
        width={this.state.contentWidth}
      />
    </div>
  },

  handleResize() {
    var contentNode = React.findDOMNode(this.refs.content);
    this.setState({ contentWidth: contentNode.clientWidth });
  },

});
```

In this case, what should the initial value of `this.state.width` be, before `componentDidMount()`? While it "works" to set it to a random non-zero positive integer, that's really just a hack where we're relying on the speed of react to re-render quickly as soon as `componentDidMount()` fires.

##### Solution one: Handle zero or null values within `building-os-charts` internals:

``` javascript
if (this.props.width !> 0) {
  noop
}
```

If width is zero, it makes sense for nothing to happen, because nothing can be rendered. I believe this should eliminate any weird React errors like above from being thrown.

##### Solution two: Document how library consumers should deal with unknown width using ternary or if else on `this.state`:

``` javascript
render() {
  <div ref="content">
    {this.state.contentWidth > 0 ? <TimeseriesChart
      height={160}
      groups={groups}
      margins={21}
      width={this.state.contentWidth}
    /> : null}
  </div>
}
```

**Comment by irvinebroque** [comment]
My other thought (haven't yet tried, might be dragons) is whether width can be handled entirely internally to `timeseries-chart.js` The chart has to be a block-level element anyways, so it just takes the full width of its parent. Same pattern as above:

1) Initial render = render empty div
2) componentDidMount = how wide am I? Set my width as state, setup event listeners.
3) second render = render the chart

**Comment by zoopoetics** [comment]
i like the idea of intuiting the width (and the height?) so you don't have to pass that manually. will make an issue.

**Comment by zoopoetics** [comment]
this was fixed in #6
